sonatype-idea.xml

```diff
@@ -22,13 +22,9 @@
   <option name="IMPORT_LAYOUT_TABLE">
     <value>
       <package name="java" withSubpackages="true" static="false" />
-      <emptyLine />
```

Comment:
They are not equivalent.  I have removed the Checkstyle check that would enforce their consistency, but still followed that convention here.
Without a Checkstyle rule, if we want a Sonatype-custom convention, perhaps we should document it explicitly.

We could
1. Revert this file and adopt the eclipse IDE config as convention.  In my view, this is inconsistent with sources in the readme such as Elements of Java Style, since it makes arbitrary, rather than logical groupings.
2. Document the old Checkstyle custom import order as convention, but remove the build-breaking check and tolerate some extra line breaks from Eclipse
3. Adopt the [Google style](https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing), grouping only static and non-static.  Google style guide is mentioned in the Readme, is more minimal, and could be configured consistently in Checkstyle, IDEA, and Eclipse config

Reply:
The ultimate reason for this repository and its content is to make codestyle a no-brainer for our developers. That means aiming for:
a) folks can rely on their IDEs to yield the expected style
b) different IDEs co-exist gracefully, without one reverting the changes from the other
c) CI can report most violations to save developers from having to point them out during reviews

The goal is in particular not to just match whatever other codestyles exist out in the world and might be considered hip/popular.

> Revert this file and adopt the eclipse IDE config as convention.

For a PR that per its title sets out to just update Checkstyle, this seems like the wise choice, i.e. keep the status quo.

If you want to rock the boat on something the entire developer org uses, wider discussions/review would be needed than just the few people you explicitly requested review from.

Reply:
I would tend towards adopting option 3 myself, but it is more drastic and has the potential to break _a lot_ of builds (even if it's easy to fix).  It might be too big of a change to make without getting larger buy-in from the development org.

I feel like there's always going to be tension between eclipse / IDEA / checkstyle, as they all have slightly different ways of doing a variety of things and it's going to be impossible to get them to all agree 100% of the time.  The approach we've taken on the data identity team is to take our checkstyle config as gospel, and if IDEA (we have no eclipse users on our team) disagrees, then we just deal with it.  It's not ideal, but everyone knows what to expect.  That is to say, in this case, if we kept the old checkstyle convention as a rule, we can make IDEA work with it, and eclipse users might have to do some extra work to come into compliance.

Reply:
I also prefer 3, but agree that a _new_ build-breaking check should not be part of this change.

So, I think we are agreed in that much, and it is just a question of whether IDE configs should stay as-is with the Checkstyle rule removed—not exactly status-quo—or I could change them to group static and non-static only, leaving Checkstyle enforcement to a future PR (if ever).
Either way, I'd like to note the convention in the readme so that we don't have to wonder which IDE config is king, the next time some discrepancy between them is found.

> If you want to rock the boat on something the entire developer org uses, wider discussions/review would be needed than just the few people you explicitly requested review from.

> It might be too big of a change to make without getting larger buy-in from the development org.

I'll surface this in \#engineering